apps/login/src/lib/service.ts

```diff
@@ -108,3 +109,20 @@ export function getServiceUrlFromHeaders(headers: ReadonlyHeaders): {
     serviceUrl: instanceUrl,
   };
 }
+
+export function constructUrl(request: NextRequest, path: string) {
+  const forwardedProto = request.headers.get("x-forwarded-proto") ?? "https";
```

Comment:
```suggestion
export function constructUrl(request: NextRequest, path: string) {
  const forwardedProto = request.headers.get("x-forwarded-proto") ? `${request.headers.get("x-forwarded-proto")}:` request.protocol;
```

Reply:
request.protocol does not exist.. we have to use request.nextUrl.protocol